adminapi/zsh-completion/adminapi

```diff
@@ -0,0 +1,62 @@
+# Copyright © 2021, InnoGames GmbH
+# Query for all hosts from serveradmin API
+# This file should be put to the directory in ${FPATH},
+#   Arch Linux, Fedora: /usr/share/zsh/site-functions
+#   Debian: /usr/share/zsh/vendor-functions
+#
+# Configuration:
+# _SERVERADMIN_HOSTS_ENABLED: if the completion should be enabled
+# _SERVERADMIN_HOSTS_QUERY: customize hosts list, string
+# _SERVERADMIN_HOSTS_TIMEOUT: customize serveradmin query timeout, integer, seconds
+# _SERVERADMIN_HOSTS_UPDATE: invalidate cache interval, integer, minutes
+# _SERVERADMIN_HOSTS_ZSTYLE: change autocompletion behavior for hosts, bool
+function _serveradmin_hosts() {
+  # Not enable if adminapi is not found
+  command -v adminapi 1>/dev/null || return
+  local _query=${_SERVERADMIN_HOSTS_QUERY:-'servertype=any(blade_enclosure 
```

Comment:
The servertypes are specific to our data model. Can we get rid of them ?

Reply:
Can you give me a default server types list, maybe?

Reply:
There are none. We ship with no default data model. What is the reason you specified them in the first place at all - performance ?

Reply:
this is the list of hosts, that will be used for autocompletion. It doesn't make sense to omit it completely, so the list reflects the internal needs.

Reply:
I don't like the hard coded list because it renders it useless for anybody else unless you know you need to adjust it. Can we maybe provide this information via a configuration file in the home directory or environment variable and include it optionally ?

Reply:
But it's the default, not the hardcoded query. One can define `_SERVERADMIN_HOSTS_QUERY`, then it will be used instead of the default value after `:-` mark.

This is from bash manual, behavior in zsh is the same:

```
${parameter:-word}
       Use Default Values.  If parameter is unset or null, the expansion of word is substituted.  Otherwise, the value of parameter is substituted.
```

Reply:
We can set the default to `not(empty(intern_ip))`. I think it's a good enough example.

Reply:
> We can set the default to not(empty(intern_ip)). I think it's a good enough example.

Sounds good. It looks reasonable  
`adminapi 'intern_ip=not(empty())'  0.43s user 0.08s system 19% cpu 2.628 total`

Reply:
Yes that sounds good.